src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,43 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+
+        List<? extends Machine> types = new ArrayList<>();
+        if (Bulldozer.class.equals(type)) {
+            types = new BulldozerProducer().get();
+        }
+        if (Excavator.class.equals(type)) {
+            types = new ExcavatorProducer().get();
+        }
+        if (Truck.class.equals(type)) {
+            types = new TruckProducer().get();
+        }
+        return new ArrayList<>(types);
```

Comment:
You're creating new producer instances each time the method is called. I'd recommend creating a map with machine type as a key and producer instance as a value. Or you can keep your solution, but let's return the result directly without the use of this 'types' variable.

Reply:
I have improved my solution with hash map as you advised. But for test passing it was obligatory to leave this line of code with new array, because I have to pass new empty list by default. Hope now it is fine.)